tools/clang/unittests/HLSL/ExtensionTest.cpp

```diff
@@ -1281,6 +1289,31 @@ TEST_F(ExtensionTest, ResourceExtensionIntrinsic) {
   VERIFY_IS_TRUE(regex.match(disassembly));
 }
 
+ TEST_F(ExtensionTest, CustomLoadOp) {
+  Compiler c(m_dllSupport);
+  c.RegisterIntrinsicTable(new TestIntrinsicTable());
+  c.Compile(
+    "float2 main(uint2 v1 : V1) : SV_Target {\n"
+    "  return CustomLoadOp(uint2(1, 2));\n"
+    "}\n",
+    { L"/Vd" }, {}
+  );
+  std::string disassembly = c.Disassemble();
+
+  // Things to check
+  // - return type is 2xfloat
+  VERIFY_IS_TRUE(
+    disassembly.npos !=
+    disassembly.find(std::string("%1 = call { float, float } @CustomLoadOp(i32 21, <2 x i32> <i32 1, i32 2>)")));
```

Comment:
Why do we convert the output to a struct, but leave the argument as a vector? Seems like it should handle them consistently.

Reply:
Input arguments for these custom intrinsics are now converted to structs (and not exploded)